reframe/frontend/cli.py

```diff
@@ -351,8 +351,9 @@ def main():
         # Filter checks by prgenv
         if not options.skip_prgenv_check:
             checks_matched = filter(
-                lambda c: c if all(c.supports_environ(e)
-                                   for e in options.prgenv) else None,
+                lambda c: c if (c.valid_prog_environs
+                                and all(c.supports_environ(e)
+                                        for e in options.prgenv)) else None,
```

Comment:
This lambda gets a bit too complicated. Since this operation is quite useful, I suggest writing a function in utility doing that thing:

```python
def allx(iterable):
    # we should check here specifically if `iterable` is actually an iterable, as real all() does
    if not iterable:
        return False

    return all(iterable)
```

Would it also make sense to provide a deferrable version of this function in this PR, too?